modules/core/src/main/java/org/apache/ignite/internal/lock/AutoLockup.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.lock;
+
+/**
+ * Represents a lockup (this is an aquisition and owning of a lock like {@link java.util.concurrent.locks.Lock})
```

Comment:
Because it is about locking and not about removing an exception

Reply:
I see, but why is it about locks and not about closeables that don't throw exceptions on closing?

Reply:
Because in this case I want to emphasize that it's about locks. Consider the code:

```
try (AutoLockup lockup = acquireLock()) {
   ...
}
```

vs

```
try (QuietAutoCloseable ac = acquireLock()) {
   ...
}
```

First one communicates clearly (using the type) that it's about a lock. Second is a bit less clear.

Please note that the fact that `close()` does not declare any exception in its `throws` is a tiny pleasantry and not the goal here, the goal is to try to use this idiom for locking. Hence the naming.